### ISSUE TITLE
EventEmitter for Model class

### ISSUE THREAD

**Issue by kulakowka** [balderdashy/waterline#1234]
Sometimes it is necessary to subscribe to certain events that occur with models. It is not always convenient to do in `afterCreate` сallback in the model file. I think it would be great to be able to subscribe to certain events (**from any part of the application**). Here's how it might look.

```javascript
User.on('created', function callback (user) {
    // Callback is invoked after you create a new instance of the model.
})

User.on('updated', function callback (user) {
    // Callback is invoked after the instance is updated an existing model.
})

User.on('deleted', function callback (user) {
    // Callback is invoked after an existing model instance will be deleted
})
```

Events are often used in JavaScript. This is a typical scenario for Node.js. I could try to implement this behavior and make the pull request. But first I would like to know the opinion of other waterline users.

What do you think about this? This ability will be useful in your applications?

**Comment by particlebanana** [comment]
@kulakowka could you add this feature to the backlog section of the [Roadmap](https://github.com/balderdashy/waterline/blob/master/ROADMAP.md) so that it can be tracked?

**Comment by luislobo** [comment]
+1

**Comment by sailsbot** [comment]
Thanks for posting, @kulakowka.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
Has this been placed into the backlog?
